ext/standard/php_random.h

```diff
@@ -0,0 +1,46 @@
+/*
+   +----------------------------------------------------------------------+
+   | PHP Version 7                                                        |
+   +----------------------------------------------------------------------+
+   | Copyright (c) 1997-2015 The PHP Group                                |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | http://www.php.net/license/3_01.txt                                  |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+   | Authors: Sammy Kaye Powers <me@sammyk.me>                            |
+   +----------------------------------------------------------------------+
+*/
+
+/* $Id$ */
+
+#ifndef PHP_RANDOM_H
+#define PHP_RANDOM_H
+
+PHP_FUNCTION(random_bytes);
+PHP_FUNCTION(random_int);
+
+PHP_MINIT_FUNCTION(lcg);
+
+ZEND_BEGIN_MODULE_GLOBALS(random)
+	int fd;
+ZEND_END_MODULE_GLOBALS(random)
+
+#ifdef ZTS
+# define RANDOM_G(v) TSRMG(random_globals_id, zend_random_globals *, v);
```

Comment:
Also ZEND_TSRMG should be used to avoid abundant function calls. But even without it I guess the TS build is broken as there's no code allocating/deallocating globals. You can lookup how it is done in some other submodule, fe browser cap.

I'd also like to ask - shouldn't the secure device actually be reopened on each request rather than once on the start? Maybe it were better to introduce the request handlers, if it should. Or even not putting it into globals, but reopening every time it's required - then it's guaranteed to be valid every time it's used.